ravenframework/SupervisedLearning/SupervisedLearning.py

```diff
@@ -208,17 +210,31 @@ def readAssembledObjects(self):
     """
     pass
 
-  def train(self, tdict, indexMap=None):
+  def train(self, trainingData, indexMap=None):
     """
       Method to perform the training of the SupervisedLearning algorithm
-      NB.the SupervisedLearning object is committed to convert the dictionary that is passed (in), into the local format
-      the interface with the kernels requires. So far the base class will do the translation into numpy
+      @ In, trainingData, dataset or dict, training dictionary
+      @ In, indexMap, dict, mapping of variables to their dependent indices, if any
+      @ Out, None
+    """
+    if self.needsDictTraining:
+      self.trainOnDictionary(trainingData, indexMap)
+    else:
+      self.amITrained = True
+      self.muAndSigmaFeatures = dict((f, (0,1)) for f in self.features)
```

Comment:
These lines is not clear to me. When dataset is needed, I do not see a training process for the ROM. Could you explain it?

Reply:
I agree, was a line missed from the old PR? If I recall correctly, we were directly overloading the "train" method if `self.needsDictTraining` is `False`.